### PR TITLE
openshift-tests: add run-resourcewatch command

### PR DESCRIPTION
**Pull request by mfojtik**
This command will start watching changes in `*.config.openshift.io` resources (including all cluster operators) and record all changes into local Git repository as a "commit".

Usage:

```
$ openshift-tests run-resourcewatch
```

If running inside the pod, it will use the SA token of the pod.

Additionally, it will produce Prometheus metrics that will contain timeline of changes in clusteroperator resource status:

```
# HELP openshift_ci_monitor_operator_cluster_operator_status [ALPHA] A metric that tracks individual cluster operator status.
# TYPE openshift_ci_monitor_operator_cluster_operator_status gauge
openshift_ci_monitor_operator_cluster_operator_status{condition="Available",name="authentication",status="True"} 1.586055762e+09
openshift_ci_monitor_operator_cluster_operator_status{condition="Available",name="cloud-credential",status="True"} 1.586054345e+09
```

Follow ups:

* Wire this into jobs that run openshift-tests (e2e-aws, etc..)
* Make sure we archive the Git repository and copy it into artifacts
* Figure out if we can make service monitor and the aggregated prometheus to scrape the metrics from this operator.

More details: https://github.com/mfojtik/ci-monitor-operator

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -2213,7 +2312,7 @@ imports:
   - pkg/apis/deviceplugin/v1beta1
   - pkg/apis/pluginregistration/v1
 - name: k8s.io/kubernetes
-  version: 5227b6f968eb46decfbb2d42e9827448a9b3910d
+  version: 3ee09fda8c72ef2489803eb69faec1e1f497302d
```
Comment (tnozicka) [inline glide.lock:2315]: you can't do that except for rebases

Comment (tnozicka) [reply glide.lock:2315]: To be more precise the vendor changes should be empty if publishing bot works. I am not sure if the bot also re-bumps origin glide to published levels, so this isolated change (without the vendor/k8s.io changes) might be fine.

**Comment by deads2k** [comment]
yeah, you changed vendored k8s by accident. don't mess with that.

**Comment by mfojtik** [comment]
@deads2k publishing bot was broken, i'm revendoring now.

**Comment by mfojtik** [comment]
@deads2k @damemi this is ready for review

File: pkg/monitor/resourcewatch/controller/configmonitor/crd_controller.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+				gvk := schema.GroupVersionKind{
+					Group:   gv.Group,
+					Version: gv.Version,
+					Kind:    crd.Spec.Names.Kind,
+				}
+				if currentKinds.Has(gvk.Kind) {
+					continue
+				}
+				currentKinds.Insert(gvk.Kind)
+				currentConfigResources = append(currentConfigResources, gvk)
```
Comment (damemi) [inline pkg/monitor/resourcewatch/controller/configmonitor/crd_controller.go:86]: nit: you could just condense these into a `if !currentKinds.Has(gvk.Kind)` block and drop the `continue`

**Comment by damemi** [comment]
/retitle Bug 1842631: openshift-tests: add run-resourcewatch command

**Pull request merged**